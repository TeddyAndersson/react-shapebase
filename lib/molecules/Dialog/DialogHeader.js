import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Heading from '../../elements/Heading';

const Element = (props) => {
    const { className, title } = props
    return(
        <header className={className}>
            <Heading size={2} scale={1.2}>
                {title}
            </Heading>
        </header>
    );
}

const DialogHeader = styled(Element)`
    padding: 0.5em;
    margin: 0;
`;

export default DialogHeader;
