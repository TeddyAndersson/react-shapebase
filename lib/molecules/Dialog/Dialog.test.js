import React from 'react';
var ReactDOM = require('react-dom')
import Dialog from './Dialog'
import renderer from 'react-test-renderer';
ReactDOM.createPortal = node => node

const lorem = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."

test('should render Dialog.Header', () => {
  const component = renderer.create(
    <Dialog.Header title={"Hey i'm a dialog!"}/>
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
})

test('should render Dialog.Content', () => {
  const component = renderer.create(
    <Dialog.Content text="And I have some dialog content"/>
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
})

test('should render Dialog.Actions', () => {
  const component = renderer.create(
    <Dialog.Actions affirmative dismissive />
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
})

test('should render Dialog', () => {
  const component = renderer.create(
    <Dialog>
      <Dialog.Header title={"Hey i'm a dialog!"}/>
      <Dialog.Content text="And I have some dialog content"/>
      <Dialog.Actions affirmative dismissive />
    </Dialog>
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
})
