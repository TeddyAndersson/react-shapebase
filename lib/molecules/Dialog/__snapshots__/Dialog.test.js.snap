// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Dialog 1`] = `
Array [
  <div
    aria-hidden="true"
    className="sc-EHOje ifNJRA"
    onClick={[Function]}
  />,
  <div
    className="sc-gzVnrw ljIbNP"
  >
     
    <div
      className="sc-jzJRlG cLhrnS sc-htpNat fLfkdc sc-bwzfXH iRJfEt"
    >
      <header
        className="sc-dnqmqq eWlMhx"
      >
        <h2
          className="sc-htoDjs eiMTPU"
        >
          Hey i'm a dialog!
        </h2>
      </header>
      <div
        className="sc-gZMcBi jVOSnQ"
      >
        <div
          className="sc-iwsKbI McfkQ"
        >
          <div
            className="sc-jTzLTM ekKGkL sc-gqjmRU hVVkNI"
          >
            <p
              className="sc-VigVT cjUJUz"
            >
              And I have some dialog content
            </p>
          </div>
        </div>
      </div>
      <footer
        className="sc-fjdhpX bgZqJX"
      >
        <div
          className="sc-gZMcBi YVBNn"
        >
          <div
            className="sc-bdVaJa fLeRSV sc-htpNat fLfkdc sc-bwzfXH iRJfEt"
            color="primary"
            disabled={false}
            href={undefined}
            onClick={[Function]}
            role="button"
          >
            <span>
               
              Cancel
               
            </span>
          </div>
          <div
            className="sc-bdVaJa fwvoUB sc-htpNat fLfkdc sc-bwzfXH iRJfEt"
            color="primary"
            disabled={false}
            href={undefined}
            onClick={[Function]}
            role="button"
          >
            <span>
               
              Agree
               
            </span>
          </div>
        </div>
      </footer>
    </div>
     
  </div>,
]
`;

exports[`should render Dialog.Actions 1`] = `
<footer
  className="sc-fjdhpX bgZqJX"
>
  <div
    className="sc-gZMcBi YVBNn"
  >
    <div
      className="sc-bdVaJa fLeRSV sc-htpNat fLfkdc sc-bwzfXH iRJfEt"
      color="primary"
      disabled={false}
      href={undefined}
      onClick={[Function]}
      role="button"
    >
      <span>
         
        Cancel
         
      </span>
    </div>
    <div
      className="sc-bdVaJa fwvoUB sc-htpNat fLfkdc sc-bwzfXH iRJfEt"
      color="primary"
      disabled={false}
      href={undefined}
      onClick={[Function]}
      role="button"
    >
      <span>
         
        Agree
         
      </span>
    </div>
  </div>
</footer>
`;

exports[`should render Dialog.Content 1`] = `
<div
  className="sc-gZMcBi jVOSnQ"
>
  <div
    className="sc-iwsKbI McfkQ"
  >
    <div
      className="sc-jTzLTM ekKGkL sc-gqjmRU hVVkNI"
    >
      <p
        className="sc-VigVT cjUJUz"
      >
        And I have some dialog content
      </p>
    </div>
  </div>
</div>
`;

exports[`should render Dialog.Header 1`] = `
<header
  className="sc-dnqmqq eWlMhx"
>
  <h2
    className="sc-htoDjs eiMTPU"
  >
    Hey i'm a dialog!
  </h2>
</header>
`;
